build.gradle.kts

```diff
@@ -30,13 +33,40 @@ allprojects {
     }
 }
 
+allOpen {
+    annotation("org.openjdk.jmh.annotations.State")
+    annotation("org.openjdk.jmh.annotations.BenchmarkMode")
+}
+
+sourceSets.create("benchmark").java {
+    srcDir("src/benchmark/kotlin")
+}
+
+val benchmarkImplementation by configurations.getting {
+    extendsFrom(configurations.implementation.get())
+}
+
+benchmark {
+    targets {
+        register("benchmark")
+    }
+
+    // specify via regex to run specific classes, e.g
+    //configurations["main"].include("BindBenchmark|michaelbull.result.FlatMapBenchmark")
+}
+
 dependencies {
     implementation(kotlin("stdlib"))
     testImplementation("junit:junit:4.13")
     testImplementation(kotlin("test-common"))
     testImplementation(kotlin("test-annotations-common"))
     testImplementation(kotlin("test-junit"))
     testImplementation(kotlin("test"))
+
+    benchmarkImplementation(sourceSets.main.get().output + sourceSets.main.get().runtimeClasspath)
+    benchmarkImplementation("org.jetbrains.kotlinx:kotlinx.benchmark.runtime-jvm:0.2.0-dev-7")
+    benchmarkImplementation("com.natpryce:result4k:2.0.0")
```

Comment:
I don't really want to benchmark against other implementations at this stage, but rather provide some benchmarks as to how people should expect *this* library to perform, and allow us to benchmark our own solutions (e.g. binding vs flatMap).

Reply:
that's fair enough!
I guess my reasoning was if you ever roadshow this library there will be the inevitable question from audience regarding how this fairs against the other implementations out there. Maybe those benchmarks belong in a separate repo however.
I'll remove that and the result4k benchmark from the pr